core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala

```diff
@@ -308,6 +309,14 @@ class HadoopRDD[K, V](
     // Do nothing. Hadoop RDD should not be checkpointed.
   }
 
+  override def persist(storageLevel: StorageLevel): this.type = {
+    if (storageLevel.deserialized) {
+      throw new SparkException("Can't cache HadoopRDDs as deserialized objects because Hadoop's" +
```

Comment:
Are there any conceivable situations where existing code that's doing this wouldn't have a bug?

> On Jan 30, 2015, at 6:59 PM, Reynold Xin <notifications@github.com> wrote:
> 
> In core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala:
> 
> > @@ -308,6 +309,14 @@ class HadoopRDD[K, V](
> >      // Do nothing. Hadoop RDD should not be checkpointed.
> >    }
> >  
> > +  override def persist(storageLevel: StorageLevel): this.type = {
> > +    if (storageLevel.deserialized) {
> > +      throw new SparkException("Can't cache HadoopRDDs as deserialized objects because Hadoop's" +
> throwing an exception is a bit extreme and will break existing code. How about just a warning?
> 
> —
> Reply to this email directly or view it on GitHub.
>